napari/layers/points/points.py

```diff
@@ -296,25 +305,20 @@ def __init__(
         self._data = np.asarray(data)
 
         # Save the properties
-        if properties is None:
+        if self.data.size == 0 and properties:
+            warnings.warn(
+                trans._(
+                    "Property_choices should be passed as property_choices, not properties. "
+                    "Workaround will be removed in 4.11"
+                ),
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            property_choices = properties
             properties = {}
-            self._properties = properties
-            self._property_choices = properties
-        elif len(data) > 0:
-            properties, _ = dataframe_to_properties(properties)
-            self._properties = self._validate_properties(properties)
-            self._property_choices = {
-                k: np.unique(v) for k, v in properties.items()
-            }
-        elif len(data) == 0:
-            self._property_choices = {
-                k: np.asarray(v) for k, v in properties.items()
-            }
-            empty_properties = {
-                k: np.empty(0, dtype=v.dtype)
-                for k, v in self._property_choices.items()
-            }
-            self._properties = empty_properties
+        self._properties, self._property_choices = self._prepare_properties(
+            properties, property_choices, True
```

Comment:
Can you make whatever `True` is here a keyword-only argument? This signature is not possible to interpret without going to the definition.